### PR TITLE
Mystical Pumpkins Integration

### PR DESCRIPTION
**Pull request by GiantLuigi4**
changes to allow for mystical pumpkins integration, however I'll leave implementing models and loot tables and whatnot

### CODE REVIEW

File: src/main/java/com/dicemc/corruptedlands/blocks/mystical_pumpkins/CorruptedPumpkinBlock.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+public class CorruptedPumpkinBlock extends MysticalPumpkinBlock implements ICorrupted, IPreventCorruption {
+	@Override
+	public void randomTick(BlockState state, ServerWorld worldIn, BlockPos pos, Random random) {
+		double threshold = random.nextDouble();
+		if (threshold >= Config.SPREAD_RATE.get()) CorruptedLandMod.Core.corruptNeighbors(pos, worldIn);
+	}
+	
+	@Override
+	public void onEntityWalk(World worldIn, BlockPos pos, Entity entityIn) {
```
Comment (Caltinor) [inline src/main/java/com/dicemc/corruptedlands/blocks/mystical_pumpkins/CorruptedPumpkinBlock.java:28]: this is unnecessary if unless pumpkins should cause damage like magma blocks do.

File: src/main/java/com/dicemc/corruptedlands/Registration.java [hunk 2]
```diff
@@ -148,4 +155,17 @@ public static void doDispense(World worldIn, ItemStack stack, int speed, Directi
 			protected void playDispenseSound(IBlockSource source) {source.getWorld().playEvent(1000, source.getBlockPos(), 0);}
 			protected void spawnDispenseParticles(IBlockSource source, Direction facingIn) {source.getWorld().playEvent(2000, source.getBlockPos(), facingIn.getIndex());}
 	    }
+
```
Comment (Caltinor) [inline src/main/java/com/dicemc/corruptedlands/Registration.java]: As a matter of preference, move these above the dispenser class and the mapBlockPairs method

**Review by Caltinor — CHANGES_REQUESTED** [review]
This was a quick review.  It looks mostly good.

File: src/main/java/com/dicemc/corruptedlands/blocks/mystical_pumpkins/CorruptedPumpkinBlock.java (hunk 1 above)
Comment (GiantLuigi4) [reply src/main/java/com/dicemc/corruptedlands/blocks/mystical_pumpkins/CorruptedPumpkinBlock.java:28]: this check is for all blocks, and you don't have any corrupted magma as well as me not knowing what that specific method did
remove it if ya want ig

**Review by Caltinor — APPROVED** [review]

**Review by Caltinor — APPROVED** [review]

**Pull request closed without merging**